scripts/verify_build.sh

```diff
@@ -36,15 +36,30 @@ curl "https://packages.chef.io/manifests/dev/automate/latest.json" > results/dev
 curl "https://packages.chef.io/manifests/current/automate/latest.json" > results/current.json
 curl "https://packages.chef.io/manifests/acceptance/automate/latest.json" > results/acceptance.json
 
-mapfile -t changed_components < <(./scripts/changed_components.rb)
-if [[ ${#changed_components[@]} -ne 0 ]]; then
+declare -a changed_components
+
+if [[ "$BUILD_ALL" = "true" ]]; then
+    for d in components/*/; do
+        if [[ -f "$d/habitat/plan.sh" ]]; then
+            changed_components+=("$d")
+        fi
+    done
+
     cat << EOF | buildkite-agent annotate --style "info"
+This change rebuilds ALL components because BUILD_ALL=$BUILD_ALL:
+$(printf '* %s\n' "${changed_components[@]}")
+EOF
 
+else
+    mapfile -t changed_components < <(./scripts/changed_components.rb)
+    if [[ ${#changed_components[@]} -ne 0 ]]; then
+        cat << EOF | buildkite-agent annotate --style "info"
```

Comment:
[nit] Let's free that cat:
```suggestion
        buildkite-agent annotate --style "info" <<EOF
```
(I'm aware that this was just moved, feel free to ignore. 😉 )